lib/kafo/kafo_configure.rb

```diff
@@ -375,11 +399,83 @@ def unset
 
     def config_file
       return CONFIG_FILE if defined?(CONFIG_FILE) && File.exists?(CONFIG_FILE)
+      return select_scenario(CONFIG_DIR) if defined?(CONFIG_DIR) && Dir.exists?(CONFIG_DIR)
       return '/etc/kafo/kafo.yaml' if File.exists?('/etc/kafo/kafo.yaml')
       return "#{::RbConfig::CONFIG['sysconfdir']}/kafo/kafo.yaml" if File.exists?("#{::RbConfig::CONFIG['sysconfdir']}/kafo/kafo.yaml")
       File.join(Dir.pwd, 'config', 'kafo.yaml')
     end
 
+    def select_scenario(config_dir)
+      # try scenario provided in the args via -S or --scenario
+      parsed = ARGV.join(" ").match /(--scenario|-S)(\s+|[=]?)(\S+)/
+      if parsed
+        scenario_file = File.join(config_dir, "#{parsed[3]}.yaml")
+        return scenario_file if File.exists?(scenario_file)
+        puts "Scenario (#{scenario_file}) was not found, can not continue"
+        self.class.exit(:unknown_scenario)
+      end
+
+      # try last_scenario.yaml
+      return self.class.prev_config_file if self.class.prev_config_file
+
+      # assume that in CONF_DIR *.yaml that has key :name is scenario definition
+      scenarios = Dir.glob(File.join(CONFIG_DIR, '*.yaml')).inject({}) do |scns, scn_file|
+        content = YAML.load_file(scn_file)
```

Comment:
load_file can raise exceptions if some file contains syntax error, so it should be rescued